pkg/sensors/load.go

```diff
@@ -340,7 +340,7 @@ func createDir(bpfDir, mapDir string) {
 
 func UnloadAll(bpfDir string) {
```

Comment:
It looks like UnloadAll() is mostly called from test programs, but is also called from RemovePrograms in observer.go. This function (IIRC) doesn't call the unloadHook on the sensors (because it unloads programs rather than sensors). Suggest we move this to tests only (if actually required by tests) and prevent use by the real program, in order to force all unloads through sensor unloading which calls the hook. The idea is that sensors have more going on than just loaded programs, and they need stopping properly on unload.

Reply:
I think we should probably remove UnloadAll(). It was added as a way to remove programs before we had the unload call, IIRC. It should only be called when the agent exits, so not running the unloadHook is probably fine, but I don't think we need it for neither the tests nor the agent shutdown anymore.